mlrun/runtimes/package/context_handler.py

```diff
@@ -0,0 +1,709 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import inspect
+import os
+import shutil
+from collections import OrderedDict
+from enum import Enum
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Type, Union
+
+import cloudpickle
+import numpy as np
+import pandas as pd
+
+from mlrun.datastore import DataItem
+from mlrun.errors import MLRunInvalidArgumentError, MLRunRuntimeError
+from mlrun.execution import MLClientCtx
+from mlrun.utils import logger
+
+
+# TODO: Move the `ArtifactType` to constants.py
+class ArtifactType(Enum):
+    """
+    Possible artifact types to log using the MLRun `context` decorator.
+    """
+
+    # Types:
+    DATASET = "dataset"
+    DIRECTORY = "directory"
+    FILE = "file"
+    OBJECT = "object"
+    PLOT = "plot"
+    RESULT = "result"
+
+    # Constants:
+    DEFAULT = RESULT
+
+
+class InputsParser:
+    """
+    A static class to hold all the common parsing functions - functions for parsing MLRun DataItem to the user desired
+    type.
+    """
+
+    @staticmethod
+    def parse_pandas_dataframe(data_item: DataItem) -> pd.DataFrame:
```

Comment:
Just wondering - is there any advantage for creating this class and a bunch of static methods that all only work on `DataItem`, rather than adding all these methods as functions of `DataItem` directly? 
Instead of `InputsParser.parse_numpy_array(data_item)` simply call `data_item.as_numpy_array()` (note the name change as well).
It also makes `parse_pandas_dataframe` redundant since you can simply call `data_item.as_df()`.

Reply:
from the PR's Future Work description: 
* Create Packager abstract class to pack (log artifact) and unpack (parse data item) python objects and files. Users will be able to write their own packagers in their projects and MLRun will know to use them when logging and parsing objects. This will replace the general OutputsLogger and InputsParser classes.

I don't want to modify the DataItem as it will be a never ending task, there are infinite objects types and the packagers we discussed is something I really like as an elegant solution. It will make DataItem an inner class of MLRun and users will simply start to use the packagers.